test/DomainRegistryV2.ts

```diff
@@ -0,0 +1,246 @@
+            });
+
+            ['google.com', 'http://new.business.com', 'https://stg0.new.com.ua', 'demo.stg0.new.com.ua'].forEach((domain) => {
+                it(`Should not allow to register ${domain} if parent domain doesn't exists`, async function () {
+                    const { domainsContract } = contractState;
+
+                    domainsContract.registerDomain(domain, { value: ether })
+
+                    await expect(domainsContract.registerDomain(domain, { value: ether }))
+                        .to.be.revertedWith("Parent domain doesn't exist");
+                });
+            });
+
+            [
+                { domain: 'google.com', parents: ['com'] },
+                { domain: 'http://new.business.no', parents: ['no', 'business.no'] },
+                { domain: 'https://stg0.new.net.ua', parents: ['ua', 'net.ua', 'new.net.ua'] },
+                { domain: 'demo.stg0.new.com.pl', parents: ['pl', 'com.pl', 'new.com.pl', 'stg0.new.com.pl'] },
+            ].forEach(({ domain, parents }) => {
+                it(`Should allow to register ${domain} domain if parents exist`, async function () {
+                    const { domainsContract, owner } = contractState;
+
+                    for (const parent of parents) {
+                        await domainsContract.registerDomain(parent, { value: ether });
+                    }
+                    await domainsContract.registerDomain(domain, { value: ether });
+
+                    const address = await domainsContract.getDomain(domain);
+
+                    expect(address).to.equal(owner.address);
+                });
+            });
+
+            it("Should take domain price in ether when assign domain 1 level", async function () {
+                const { domainsContract, addr1, owner } = contractState;
+
+                const tx = await domainsContract.connect(addr1).registerDomain('com', { value: ether });
+                await expect(tx).to.changeEtherBalances(
+                    [addr1, owner],
+                    [-priceLevel1Domain, priceLevel1Domain - priceLevel1Domain * BigInt(10) / BigInt(100)]
+                );
+            });
+
+            it("Should reward parent domain owner when assign domain 2 level", async function () {
+                const { domainsContract, addr1, addr2, owner } = contractState;
+
+                await domainsContract.connect(addr1).registerDomain('com', { value: ether });
+                const tx = await domainsContract.connect(addr2).registerDomain('test.com', { value: ether });
+
+                await expect(tx).to.changeEtherBalances(
+                    [addr2, addr1, owner],
+                    [
+                        -priceLevel2Domain, 
+                        priceLevel2Domain * BigInt(10) / BigInt(100),
+                        priceLevel2Domain - priceLevel2Domain * BigInt(10) / BigInt(100)
+                    ]
+                );
+            });
+        });
```

Comment:
two tests above have `Should take domain price in ether when assign domain 1 level`, `Should reward parent domain owner when assign domain 2 level` been changed since V1 version